core/eolearn/core/eodata.py

```diff
@@ -533,7 +544,7 @@ def reset_feature_type(self, feature_type: FeatureType) -> None:
         if feature_type.has_dict():
             self[feature_type] = {}
         elif feature_type is FeatureType.BBOX:
-            self[feature_type] = None
+            raise ValueError("The BBox of an EOPatch should never be undefined.")
```

Comment:
I do not know how to improve error massage just se a smal issue wit our current typing for `bbox: Optional[BBox]`

Reply:
It is a bit 'in the future'. I can change it to a warning if you think it would be better.

Reply:
if it is going to be a bit i guess it can stay, just want to point out possible "miss erroring" ...

Reply:
i will keep it as it is, because we already force BBOX to be present whenever possible. Let the users see a glimpse of the future. In the next release there will be a deprecation warning about BBox being necessary anyway.